src/lib/npfg/CourseToAirspeedRefMapper.cpp

```diff
@@ -36,42 +36,81 @@ using matrix::Vector2f;
 
 float
 CourseToAirspeedRefMapper::mapCourseSetpointToHeadingSetpoint(const float bearing_setpoint, const Vector2f &wind_vel,
-		float airspeed_true) const
+		float airspeed_max, float min_ground_speed) const
 {
-
-	Vector2f bearing_vector = Vector2f{cosf(bearing_setpoint), sinf(bearing_setpoint)};
+	const Vector2f bearing_vector = Vector2f{cosf(bearing_setpoint), sinf(bearing_setpoint)};
 	const float wind_cross_bearing = wind_vel.cross(bearing_vector);
 	const float wind_dot_bearing = wind_vel.dot(bearing_vector);
 
 	const Vector2f air_vel_ref = refAirVelocity(wind_vel, bearing_vector, wind_cross_bearing,
-				     wind_dot_bearing, wind_vel.norm(), airspeed_true);
+				     wind_dot_bearing, wind_vel.norm(), airspeed_max, min_ground_speed);
 
 	return atan2f(air_vel_ref(1), air_vel_ref(0));
 }
 
+float
+CourseToAirspeedRefMapper::getMinAirspeedForCurrentBearing(const float bearing_setpoint, const Vector2f &wind_vel,
+		float airspeed_max, float min_ground_speed) const
+{
+	const Vector2f bearing_vector = Vector2f{cosf(bearing_setpoint), sinf(bearing_setpoint)};
+	const float wind_cross_bearing = wind_vel.cross(bearing_vector);
+	const float wind_dot_bearing = wind_vel.dot(bearing_vector);
+
+	const Vector2f air_vel_ref = refAirVelocity(wind_vel, bearing_vector, wind_cross_bearing,
+				     wind_dot_bearing, wind_vel.norm(), airspeed_max, min_ground_speed);
+
+	return air_vel_ref.norm();
+}
+
 matrix::Vector2f CourseToAirspeedRefMapper::refAirVelocity(const Vector2f &wind_vel, const Vector2f &bearing_vec,
 		const float wind_cross_bearing, const float wind_dot_bearing,
-		const float wind_speed, float airspeed_true) const
+		const float wind_speed, float airspeed_max, float min_ground_speed) const
 {
 	Vector2f air_vel_ref;
 
-	if (bearingIsFeasible(wind_cross_bearing, wind_dot_bearing, airspeed_true, wind_speed)) {
-		const float airsp_dot_bearing = projectAirspOnBearing(airspeed_true, wind_cross_bearing);
-		air_vel_ref = solveWindTriangle(wind_cross_bearing, airsp_dot_bearing, bearing_vec);
+	const float airspeed_nom_ = 15.f; //TODO pass as arguemnt or remove
+
+	if (min_ground_speed > wind_dot_bearing) {
```

Comment:
@RomanBapst This part is largely copied from the previous NPFG implementation to have the minimum groundspeed handling again. I've removed the non-default NPFG options though.

Reply:
I've removed a lot of extra logic, need to check if we're having still a robust behavior in excess wind.